### DISCUSSION TITLE
Multi protocol support for single server instance

### DISCUSSION THREAD

**Discussion by jrauschenbusch** [fastify/fastify#4522] [Q&A]
Is it possible to have a single server instance supporting both, HTTP/1.1 and HTTP/2 at the same time?

Our use case is, that we want to expose a bunch of routes via HTTP/2 to our customers while exposing a /healthz endpoint also via HTTP/1.1.

Reason: Unfortunately Kubernetes currently just supports HTTP/1.1 for it's container probes (except using the command-Syntax).

**Comment by mcollina** [comment]
I have spawn a Fastify service inside a plugin in those cases.

**Reply by jrauschenbusch** [reply]
Could you provide an example for this? I wondered if the NodeJS HTTP/2 Compatibility API could solve this.

https://nodejs.org/api/http2.html#compatibility-api

> The Compatibility API has the goal of providing a similar developer experience of HTTP/1 when using HTTP/2, making it possible to develop applications that support both [HTTP/1](https://nodejs.org/api/http.html) and HTTP/2.

But overwriting the socketFactory did not worked. HTTP/1.1 is not supported then.

Side note: I don’t want to use the secure server option (where ALPN would be an option) as inside the cluster we don’t want to handle app-specific certificates. This is done via our Service mesh. So no need for extra layer of security.

**Reply by mcollina** [reply]
Check out https://github.com/platformatic/platformatic/blob/main/packages/service/lib/metrics-plugin.js
